kythe/java/com/google/devtools/kythe/platform/java/filemanager/CompilationUnitPathFileManager.java

```diff
@@ -357,7 +357,9 @@ private void setSystemOption(String value) throws IOException {
               .toAbsolutePath();
       Path systemRoot = Files.createDirectory(temporaryDirectory.resolve("system"));
       try (Stream<Path> stream = Files.walk(sys.resolve("lib"))) {
-        for (Path path : (Iterable<Path>) stream::iterator) {
+        Iterator<Path> it = stream.iterator();
+        while (it.hasNext()) {
+          Path path = it.next();
```

Comment:
As this seems objectively worse than the original, it's going to need a comment to prevent well-meaning changes from reverting this in the future.

Reply:
Sorry, "objectively worse" is harsher than intended.  Mostly meant to suggest that without a corresponding comment this looks trivially like something which should be tackled with a for loop rather than an explicit iterator and that transformation likely to be re-applied.

Reply:
Done. I think "objectively worse" is not super unfair--it's very annoying how hard it is to iterate over a stream while throwing a checked exception.